snippets/indexOfAll.md

```diff
@@ -0,0 +1,20 @@
+### indexOfAll
+
+Returns all indices of `val` in an array. If `val` never occurs, returns `[-1]`.
+
+Use `Array.forEach()` to loop over elements and `Array.push()` to store indices for matching elements.
+Return `[-1]` if `length` of the array of indices is `0`, otherwise return the array of indices.
+
+``` js
+const indexOfAll = (arr, val) => {
+  const indices = [];
+  arr.forEach((el, i) => el === val && indices.push(i))
+  return indices.length ? indices : [-1];
```

Comment:
I think you misunderstood me:
```js
return indices.length ? indices : -1;
```

Reply:
This way you either get back an array, in which case do what you need to, or a -1, no third case. This is a consistent testable API in that Array.isArray will gladly work on.

I just meant you should return either the array of indeces or the -1, not a singular value if there is only one index

Reply:
I think returning an array in all cases feels more consistent, as you can then use it as an array no matter what. Might be just me, though.

Reply:
currently
```js
indexOfAll([1,2,3], 4).filter(a => a != -1).map(console.log)
```

If this is the case, we should return an empty array, not an array of -1, otherwise the filter would be needed for any chaining

Reply:
@skatcat31 An empty array is a decent idea that improves the snippet, could you send a PR for that? (I'm on mobile currently)

Reply:
@Chalarangelo I'm in  the middle of deploying and have time sensitive issues after, so I will not be able to( don't have this repo on my work PC )

Reply:
You can edit the snippet right from the browser, @skatcat31. The main reason I suggest this is so that you can take credit for the change, which was your idea to begin with! 😉

Reply:
... I meant to just make a PR but apparently I edited it directly on master... Web apps are confusing -.-

I can't figure out how to bring my side along fork to date and create an MR from it. Is what I did an issue?( this was due to a lack of knowledge of how Github does it's built in tools )

Reply:
Nop, it's a change discussed and approved, no problem at all!